app/src/main/java/com/example/concordia_campus_guide/Activities/SearchActivity.java

```diff
@@ -107,27 +110,40 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
             }
         });
 
+        if(shouldSetNextClassClickListener){
+            nextClassArrow.setVisibility(View.VISIBLE);
+            nextClassRow.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    Place place = nextClassPlace;
+                    openRoutesPage(place);
+                }
+            });
+        }
+
+    }
+
+    private void populateNextClassString() {
         final CalendarEvent calendarEvent = calendarViewModel.getEvent(this);
         final String eventString;
         final String eventLocation;
 
-        if (calendarEvent != null) {
-            eventString = calendarViewModel.getNextClassString((calendarEvent));
-            nextClassText.setText(eventString);
+        if(calendarEvent != null){
             eventLocation = calendarEvent.getLocation();
             Place place = mViewModel.getRoomFromDB(eventLocation);
-
-            if (!eventString.equals("Event is incorrectly formatted")) {
-                if (place == null) {
-                    nextClassText.setText("Location not found");
-                } else {
-                    nextClassArrow.setVisibility(View.VISIBLE);
-                    nextClassRow.setOnClickListener(new View.OnClickListener() {
-                        @Override
-                        public void onClick(View view) {
-                            openRoutesPage(place);
-                        }
-                    });
+            if(place == null){
+                nextClassText.setText(getResources().getString(R.string.location_not_found));
+                shouldSetNextClassClickListener = false;
+            }else{
+                eventString = calendarViewModel.getNextClassString((calendarEvent));
+                nextClassText.setText(eventString);
+                // double check this value
+                if(eventString.equals(getResources().getString(R.string.incorrect_format_event))){
+                    shouldSetNextClassClickListener = false;
+                }
+                else{
+                    shouldSetNextClassClickListener = true;
+                    nextClassPlace = place;
```

Comment:
For each if else statement, can you put them into separate functions ? it would make the function cleaner. what do u think ?